nixos/modules/services/desktops/pipewire/pipewire.nix

```diff
@@ -72,15 +74,75 @@ in {
           https://github.com/PipeWire/pipewire/blob/master/NEWS
         '';
       };
+
+      config = {
+        defaultConfig = mkOption {
+          default = true;
+          type = types.bool;
+          description = lib.mdDoc ''
+            Enable default configs for PipeWire.
+          '';
+        };
+        pipewire = mkOption {
+          type = json.type;
+          default = lib.importJSON "${pipewire-json-conf}/pipewire.conf";
```

Comment:
I think you need some form of lib.recursiveUpdate here? Otherwise setting one of the sub-options seems to override all of the defaults i.e.:

```
nix-repl> :p (lib.evalModules { modules = [{options = { foo = lib.mkOption { type = json.type; def
ault = { a = 1; b = 2; };  };  };} { config.foo.b = 3; } ]; }).config
{ foo = { b = 3; }; }
```
... where

```
{ foo = { a = 1; b = 3; }; }
```